tests/test_hierarchy.cc

```diff
@@ -47,6 +47,107 @@ double Source<dim>::value(dealii::Point<dim> const &, unsigned int const) const
   return 0.;
 }
 
+template <int dim>
+class ConstantMaterialProperty : public dealii::Function<dim>
+{
+public:
+  ConstantMaterialProperty() = default;
+
+  virtual double value(dealii::Point<dim> const &p,
+                       unsigned int const component = 0) const override final;
+};
+
+template <int dim>
+double ConstantMaterialProperty<dim>::value(dealii::Point<dim> const &,
+                                            unsigned int const) const
+{
+  return 1.;
```

Comment:
If we are not using ETI inside the tests, is there a reason to move the body of the member functions out of declaration?